addons/sale/models/payment.py

```diff
@@ -61,33 +61,40 @@ def _set_transaction_pending(self):
         super(PaymentTransaction, self)._set_transaction_pending()
 
         for record in self:
-            sales_orders = record.sale_order_ids.filtered(lambda so: so.state == 'draft')
-            sales_orders.force_quotation_send()
+            sales_orders = record.sale_order_ids.filtered(lambda so: so.state in ['draft', 'sent'])
```

Comment:
Why add the 'sent' sale_orders in the domain?
That means you will send the emails for sale_order that are already sent?
And the following line that write the states as 'sent' does not make sense for sale_orders that are already sent.

Reply:
@awa-odoo 
we are not sending quotation mail here, it will send the payment confirmation mail not quotation mail.   
for portal:
   when the sales order is in the draft state we are not allowing to do a payment from portal so this method will never called for the draft sales order. 
for the website :
      the sales order is always in a draft state and we are sending payment confirmation mail instead of quotation mail(but the transaction is in a pending state so instead of confirming the order we make the order as 'sent' ). 
Thanks

Reply:
@mba-odoo  Thanks for the explanation.

Then maybe the `_send_order_confirmation_email` should take care of updating sale_orders that still have the 'draft' state to the 'sent' state instead of calling it separately?

Because before the `_set_transaction_authorized` method called the `force_quotation_send` that automatically marked the sale orders as 'sent' but I don't think it's the case anymore. Forcing the 'sent' state in the `_send_order_confirmation_email` method would make sure it's always done.

Do you follow me?

Reply:
@awa-odoo , in `_set_transaction_authorized` the order is already 'confirmed' so at that time we do not require to make it sent. while in `_set_transaction_pending' still transaction is pending so we make the order as 'sent'. 
Thanks

Reply:
@mba-odoo I still think it should be the responsibility of the _send_order_confirmation_email.
It will guarantee that we keep a coherent SO state if the method is called from somewhere else.

And it will also ease the replacement of the `force_quotation_send` method in the `test_sale_order` by your new `_send_order_confirmation_email`.

Reply:
@awa-odoo , if we move the state change related code in `_send_order_confirmation_email` we need to call this method before the `action_confirm` in `_set_transaction_authorized` and in a sent mail the buttons and other some contents are based on the state of order. so it will display the wrong content. 
Thanks

Reply:
@awa-odoo , I also prefer to keep `force_quotation_send` in the test because we almost use default template in  `force_quotation_send` and change state always as 'sent' while in `_send_order_confirmation_email` we use the template set in config parameter. Thanks